src/app/digital/internal/assembly/components/ORGateAssembler.ts

```diff
@@ -1,143 +1,142 @@
-    }
 
-    public assemble(gate: Schema.Component, ev: unknown) {
-        const transformChanged = /* use ev to see if our transform changed */ true;
-        const selectionChanged = /* use ev to see if we were de/selected */ true;
-        const portAmtChanged   = /* use ev to see if the number of ports changed */ true;
-
-        if (!transformChanged && !selectionChanged && !portAmtChanged)
-            return;
-
-        if (transformChanged) {
-            // Update transform
-            this.view.componentTransforms.set(gate.id, new Transform(
-                V(gate.props.x ?? 0, gate.props.y ?? 0),
-                this.size,
-                (gate.props.angle ?? 0),
-            ));
-        }
+        const tint = (selected ? parseColor(this.options.selectedFillColor) : undefined);
+
+        return {
+            kind: "SVG",
 
-        this.portAssembler.assemble(gate, ev);
+            svg:       "or.svg",
+            transform: this.cache.componentTransforms.get(gate.id)!,
 
-        const [prevLine, prevImg] = (this.view.componentPrims.get(gate.id) ?? []);
+            tint,
+        };
+    }
 
-        const image = [];
-        const line = this.assembleQuadCurve(gate);
-        const img  = ((!prevImg || transformChanged) ? this.assembleImage(gate) : prevImg);
+    public override assemble(gate: Schema.Component, ev: Set<AssemblyReason>) {
+        super.assemble(gate, ev);
+
+        const added            = ev.has(AssemblyReason.Added);
+        const transformChanged = ev.has(AssemblyReason.TransformChanged);
+        const portAmtChanged   = ev.has(AssemblyReason.PortsChanged);
+        const selectionChanged = ev.has(AssemblyReason.SelectionChanged);
+
+        if (added || transformChanged) {
+            this.cache.componentPrims.set(gate.id, [
+                this.assembleImage(gate),
+                ...this.assembleQuadCurve(gate),
+            ]);
+        } else if (portAmtChanged) {
+            const [img, ..._] = this.cache.componentPrims.get(gate.id)!;
+            this.cache.componentPrims.set(gate.id, [
+                img,
+                ...this.assembleQuadCurve(gate),
+            ]);
+        } else if (selectionChanged) {
+            const [img, ...quadCurves] = this.cache.componentPrims.get(gate.id)!;
+
+            if (quadCurves.some((quadCurve) => quadCurve.kind !== "QuadCurve") || img.kind !== "SVG") {
+                console.error(`Invalid prim type in ORGateAssembler! ${
+                    quadCurves.map((quadCurve) => quadCurve.kind !== "QuadCurve")
+                }, ${img.kind}`);
+                return;
+            }
 
-        line.forEach((q) => image.push(q));
-        image.push(img);
+            const selected = this.selections.has(gate.id);
 
-        this.view.componentPrims.set(gate.id, image);
+            quadCurves.forEach((quadCurve) => {
+                // TODO: better type safe way of handling this?
```

Comment:
Yeah... we need a much better system for this, I hate it so much lol; but it's the correct way for now